models/demos/falcon7b/tt/falcon_attention.py

```diff
@@ -662,7 +662,7 @@ def forward(
         # We always store max_position_embeddings for kv_cache,
         # so we need separate variable to store the actual len of the kv_cache
         assert layer_past is not None
-        assert layer_past_len > 0 and layer_past_len <= self.max_position_embeddings
```

Comment:
Why do we allow layer_past_len to be 0?

Reply:
layer_past_len == 0 is required for the first decode iteration when there have been no past tokens